tests/nn/checkpoint/test_checkpoint_activations.py

```diff
@@ -267,3 +267,38 @@ def test_deprecated_path():
     # Check if direct import works as before.
     ffn = nn.Sequential(nn.Linear(32, 128), nn.Dropout(p=0.5), nn.Linear(128, 32),)
     ffn = deprecated_checkpoint_wrapper(ffn, {})
+
+
+@skip_if_no_cuda
+def test_list_input():
+    """Test to ensure that we correctly does the checkpoint when the input is a list.
+
+    Note: Testing shows that PyTorch's torch.utils.checkpoint function does not pass this test.
```

Comment:
should we follow up about this offline?

Reply:
I am noting this most for future reference. I think it is better to focus on one checkpoint code instead of two. The other code is useful in debugging in case we found a bug only exists in this implementation not the other.  I think we might want to update the doc on the pytorch version and pointing users to use this version going forward.